#  Dataviz project
## ARTAUD Lucas & SIVASUBRAMANIAM Iswarya DIA 1

## Imports

In [97]:
import pandas as pd

from geopy.geocoders import Nominatim
import pycountry_convert as pc 

import matplotlib.pyplot as plt
import plotly.express as px
import plotly.graph_objects as go
import plotly.subplots as sp

## Dataset Analysis

In [98]:
df=pd.read_csv("Life Expectancy Data.csv")

In [99]:
df

Country  Year      Status  Life expectancy   Adult Mortality  \
0     Afghanistan  2015  Developing              65.0            263.0   
1     Afghanistan  2014  Developing              59.9            271.0   
2     Afghanistan  2013  Developing              59.9            268.0   
3     Afghanistan  2012  Developing              59.5            272.0   
4     Afghanistan  2011  Developing              59.2            275.0   
...           ...   ...         ...               ...              ...   
2933     Zimbabwe  2004  Developing              44.3            723.0   
2934     Zimbabwe  2003  Developing              44.5            715.0   
2935     Zimbabwe  2002  Developing              44.8             73.0   
2936     Zimbabwe  2001  Developing              45.3            686.0   
2937     Zimbabwe  2000  Developing              46.0            665.0   

      infant deaths  Alcohol  percentage expenditure  Hepatitis B  Measles   \
0                62     0.01               71.279624         65.0      1154   
1                64     0.01               73.523582         62.0       492   
2                66     0.01               73.219243         64.0       430   
3                69     0.01               78.184215         67.0      2787   
4                71     0.01                7.097109         68.0      3013   
...             ...      ...                     ...          ...       ...   
2933             27     4.36                0.000000         68.0        31   
2934             26     4.06                0.000000          7.0       998   
2935             25     4.43                0.000000         73.0       304   
2936             25     1.72                0.000000         76.0       529   
2937             24     1.68                0.000000         79.0      1483   

      ...  Polio  Total expenditure  Diphtheria    HIV/AIDS         GDP  \
0     ...    6.0               8.16         65.0        0.1  584.259210   
1     ...   58.0               8.18         62.0        0.1  612.696514   
2     ...   62.0               8.13         64.0        0.1  631.744976   
3     ...   67.0               8.52         67.0        0.1  669.959000   
4     ...   68.0               7.87         68.0        0.1   63.537231   
...   ...    ...                ...          ...        ...         ...   
2933  ...   67.0               7.13         65.0       33.6  454.366654   
2934  ...    7.0               6.52         68.0       36.7  453.351155   
2935  ...   73.0               6.53         71.0       39.8   57.348340   
2936  ...   76.0               6.16         75.0       42.1  548.587312   
2937  ...   78.0               7.10         78.0       43.5  547.358878   

      Population   thinness  1-19 years   thinness 5-9 years  \
0     33736494.0                   17.2                 17.3   
1       327582.0                   17.5                 17.5   
2     31731688.0                   17.7                 17.7   
3      3696958.0                   17.9                 18.0   
4      2978599.0                   18.2                 18.2   
...          ...                    ...                  ...   
2933  12777511.0                    9.4                  9.4   
2934  12633897.0                    9.8                  9.9   
2935    125525.0                    1.2                  1.3   
2936  12366165.0                    1.6                  1.7   
2937  12222251.0                   11.0                 11.2   

      Income composition of resources  Schooling  
0                               0.479       10.1  
1                               0.476       10.0  
2                               0.470        9.9  
3                               0.463        9.8  
4                               0.454        9.5  
...                               ...        ...  
2933                            0.407        9.2  
2934                            0.418        9.5  
2935                            0.427      

In [100]:
df.shape

(2938, 22)

In [101]:
df.isna().sum()

Country                              0
Year                                 0
Status                               0
Life expectancy                     10
Adult Mortality                     10
infant deaths                        0
Alcohol                            194
percentage expenditure               0
Hepatitis B                        553
Measles                              0
 BMI                                34
under-five deaths                    0
Polio                               19
Total expenditure                  226
Diphtheria                          19
 HIV/AIDS                            0
GDP                                448
Population                         652
 thinness  1-19 years               34
 thinness 5-9 years                 34
Income composition of resources    167
Schooling                          163
dtype: int64

## Context and motivation

##### The data set chosen is a statistical data on factors influencing Life Expectancy. The data comes from the World Health Organization over a 15-year period.   

##### Many studies in the past have explored the factors influencing life expectancy, centering around demographic variables, income composition, and mortality rates. However, these studies often neglected the impact of immunization and the Human Development Index. Additionally, some prior research relied on a one-year dataset for all countries but doing this research for a period of time of 15 year enables us to visualise the changes over time. This data set allows us to do a country based observation to identify the main factors that are contributing to lower life expectancy.  
##### This dataset encompasses health factors for 193 countries from 2000 to 2015, with 22 columns and 2938 rows. Our preliminary analysis indicates that the population, Hepatitis B, and GDP columns contain the majority of missing data. Rather than removing all missing values and losing valuable information, we have opted to retain the incomplete rows.

##### The project's motivation is to analyze various factors influencing life expectancy, comparing them on different scales such as continents and development statuses.
##### The objective is to gain insights into factors affecting life expectancy, guiding public health interventions and policies. Targeted healthcare initiatives could be guided, for instance, by identifying particular regions or demographic groups experiencing challenges with life expectancy. Furthermore, knowledge of how social determinants, economic variables, and immunizations affect life expectancy can support evidence-based decision-making at the national and international levels. 

#### The link : https://www.kaggle.com/datasets/kumarajarshi/life-expectancy-who

#### Columns

* **Country**: Country
* **Year**: Year
* **Status**: Classification of countries as 'developed' or 'developing' based on their gross domestic product(GDP).
* **Life expectancy**: Life expectancy (years of age)
* **Adult Mortality**: Adult Mortality Rates of both sexes (Probability of dying between 15 and 60 years per 1000 population)
* **infant deaths**: Number of Infant (0-1 year of age) Deaths per 1000 population.
* **Alcohol**: Alcohol, recorded per capita (15+) consumption (in litres of pure alcohol).
* **Percentage expenditure**: Expenditure on health as a percentage of GPD per capita. (%)
* **Hepatitis B**: Hepatitis B immunization coverage among 1-year-olds. (%)
* **Measles**: Number of reported cases per 1000 population.
* **BMI**: Average Body Mass index of entire population
* **Under-five deaths**: Number of under-five deaths per 1000 population
* **Polio**: Polio immunization coverage among 1-year-olds (%)
* **Total expenditure**: General government expenditure on health as a percentage of total government expenditure (%)
* **Diphtheria**: Diphtheria tetanus toxoid and pertussis (DTP3) immunization coverage among 1-year-olds (%)
* **HIV/AIDS**: Deaths per 1000 live births HIV/AIDS (0-4 years)
* **GDP**: Gross Domestic Product per capita (in USD)
* **Population**: Population of the country
* **Thinness 1-19 years**: Prevalence of thinness among children and adolescents for Age 10 to 19 (%) 
* **Thinness 5-9 years**: Prevalence of thinness among children for Age 5 to 9 (%)
* **Income composition of resources**: Human Development Index in terms of income composition of resources (index ranging from 0 to 1)
* **Schooling**: Number of years of Schooling (years)

## Creating new columns

#### With the library pycountry_convert we are going to create a new column "Continent" that will correspond to the continent of the country. And with Neonatim we are going to generate the latitude and longitude for each country in order to create maps.

In [102]:
continent_name = {
    'AF': 'Africa',
    'AS': 'Asia',
    'EU': 'Europe',
    'NA': 'North America',
    'OC': 'Oceania',
    'SA': 'South America',
    'AN': 'Antarctica'
}

In [103]:
# Replacing the special cases
df['Country'] = df['Country'].replace({'Bolivia (Plurinational State of)': 'Bolivia', 'Iran (Islamic Republic of)': 'Iran', 'Micronesia (Federated States of)':'Micronesia','Republic of Korea':'Korea, Republic of', 'Korea':"Korea (Democratic People's Republic of)",'The former Yugoslav republic of Macedonia':'North Macedonia','Venezuela (Bolivarian Republic of)':'Venezuela'})

In [104]:
unique_countries = pd.DataFrame(df['Country'].unique(), columns=['Country']) # Taking all the country names
geolocator = Nominatim(user_agent="my_geocoder") 

In [105]:
def get_lat_lon(country):
    location = geolocator.geocode(country, timeout=5)# locating the country
    if location:
        return pd.Series({'latitude': location.latitude, 'longitude': location.longitude}) # returning the longitude and latitude
    else:
        return pd.Series({'latitude': None, 'longitude': None})

In [106]:
def convert_continent(country):
    # convert country name to country code
    country_code =pc.country_name_to_country_alpha2(country,cn_name_format="default")
    # convert country_code to continent code
    try:
        continent_code = pc.country_alpha2_to_continent_code(country_code)
        return continent_name.get(continent_code, None)
    except :
        return None

In [107]:
unique_countries['Continent']=unique_countries['Country'].apply(convert_continent)


In [108]:
unique_countries[['latitude', 'longitude']] = unique_countries['Country'].apply(get_lat_lon)

In [109]:
unique_countries

Country      Continent   latitude   longitude
0            Afghanistan           Asia  33.768006   66.238514
1                Albania         Europe   1.231526  -75.892043
2                Algeria         Africa  28.000027    2.999983
3                 Angola         Africa -11.877577   17.569124
4    Antigua and Barbuda  North America  17.223472  -61.955461
..                   ...            ...        ...         ...
188            Venezuela  South America   8.001871  -66.110932
189             Viet Nam           Asia  15.926666  107.965086
190                Yemen           Asia  16.347124   47.891527
191               Zambia         Africa -14.518912   27.558988
192             Zimbabwe         Africa -18.455496   29.746841

[193 rows x 4 columns]

In [110]:
df = pd.merge(df, unique_countries, on='Country', how='left') # merrging the created columns

In [111]:
df

Country  Year      Status  Life expectancy   Adult Mortality  \
0     Afghanistan  2015  Developing              65.0            263.0   
1     Afghanistan  2014  Developing              59.9            271.0   
2     Afghanistan  2013  Developing              59.9            268.0   
3     Afghanistan  2012  Developing              59.5            272.0   
4     Afghanistan  2011  Developing              59.2            275.0   
...           ...   ...         ...               ...              ...   
2933     Zimbabwe  2004  Developing              44.3            723.0   
2934     Zimbabwe  2003  Developing              44.5            715.0   
2935     Zimbabwe  2002  Developing              44.8             73.0   
2936     Zimbabwe  2001  Developing              45.3            686.0   
2937     Zimbabwe  2000  Developing              46.0            665.0   

      infant deaths  Alcohol  percentage expenditure  Hepatitis B  Measles   \
0                62     0.01               71.279624         65.0      1154   
1                64     0.01               73.523582         62.0       492   
2                66     0.01               73.219243         64.0       430   
3                69     0.01               78.184215         67.0      2787   
4                71     0.01                7.097109         68.0      3013   
...             ...      ...                     ...          ...       ...   
2933             27     4.36                0.000000         68.0        31   
2934             26     4.06                0.000000          7.0       998   
2935             25     4.43                0.000000         73.0       304   
2936             25     1.72                0.000000         76.0       529   
2937             24     1.68                0.000000         79.0      1483   

      ...   HIV/AIDS         GDP  Population   thinness  1-19 years  \
0     ...        0.1  584.259210  33736494.0                   17.2   
1     ...        0.1  612.696514    327582.0                   17.5   
2     ...        0.1  631.744976  31731688.0                   17.7   
3     ...        0.1  669.959000   3696958.0                   17.9   
4     ...        0.1   63.537231   2978599.0                   18.2   
...   ...        ...         ...         ...                    ...   
2933  ...       33.6  454.366654  12777511.0                    9.4   
2934  ...       36.7  453.351155  12633897.0                    9.8   
2935  ...       39.8   57.348340    125525.0                    1.2   
2936  ...       42.1  548.587312  12366165.0                    1.6   
2937  ...       43.5  547.358878  12222251.0                   11.0   

       thinness 5-9 years  Income composition of resources  Schooling  \
0                    17.3                            0.479       10.1   
1                    17.5                            0.476       10.0   
2                    17.7                            0.470        9.9   
3                    18.0                            0.463        9.8   
4                    18.2                            0.454        9.5   
...                   ...                              ...        ...   
2933                  9.4                            0.407        9.2   
2934                  9.9                            0.418        9.5   
2935                  1.3                            0.427       10.0   
2936                  1.7                            0.427        9.8   
2937                 11.2                            0.434        9.8   

      Continent   latitude  longitude  
0          Asia  33.768006  66.238514  
1          Asia  33.768006  66.238514  
2          Asia  33.768006  66.238514  
3          Asia  33.768006  66.238514  
4          Asia  33.768006  66.238514  
...         ...        ...        ...  
2933     Africa -18.455496  29.746841  
2934     Africa -18.455496  29.746841  
2935     Africa -18.455496  29.746841  
2936     Africa -18.455496  29.746841  
29

## Visualisations

## General Analysis

### 1. Average life expectancy and population over the years

In [112]:
average_life_expectancy_yearly = df.groupby('Year')['Life expectancy '].mean().reset_index()
average_population_yearly = df.groupby('Year')['Population'].mean().reset_index()

In [141]:
fig1 = sp.make_subplots(rows=2, cols=1, subplot_titles=['Average Life Expectancy', 'Average Population'])

fig1.add_trace(
    go.Scatter(x=average_life_expectancy_yearly['Year'], y=average_life_expectancy_yearly['Life expectancy '],
               mode='lines', name='Life Expectancy'),
    row=1, col=1
)

fig1.add_trace(
    go.Scatter(x=average_population_yearly['Year'], y=average_population_yearly['Population'],
               mode='lines', name='Population'),
    row=2, col=1
)
# Update layout for better readability
fig1.update_layout(
    title_text="Average Life Expectancy and Population Over Time",
    title_font=dict(size=24),
    legend=dict(font=dict(size=16)),
    font=dict(size=15),
    width=1500,
    height=700,
)
fig1.show()

####  Between 2000 and 2015 we can see that the average life expectancy has increased from 67 to 72 years. For the average population, it is inconsistent over time, in fact we can see ups and downs, especially during 2008 and 2010.

### 2. Life Expectancy over the years of the top 5 and bottom 5 countries

In [114]:
average_life_expectancy = df.groupby('Country')['Life expectancy '].mean().reset_index() # calculate the average life expectancy
top5_countries = average_life_expectancy.nlargest(5, 'Life expectancy ') # take the top 10 average life expectancy
bottom5_countries = average_life_expectancy.nsmallest(5, 'Life expectancy ') # take the bottom 10 average life expectancy

In [115]:
# Filtering the Original DataFrame for the Selected Countries
selected_countries = top5_countries['Country'].tolist() + bottom5_countries['Country'].tolist()
filtered_df = df[df['Country'].isin(selected_countries)] 

In [116]:
selected_countries

['Japan',
 'Sweden',
 'Iceland',
 'Switzerland',
 'France',
 'Sierra Leone',
 'Central African Republic',
 'Lesotho',
 'Angola',
 'Malawi']

In [142]:
fig2 = px.line(filtered_df, x='Year', y='Life expectancy ', color='Country',title='Life Expectancy Over the Years for the top 5 and bottom 5 Selected Countries')
# Update layout for better readability
fig2.update_layout(
    xaxis_title='Year',
    yaxis_title='Life Expectancy',
    title_font=dict(size=24),
    legend_title='Country',
    font=dict(size=15),
    legend=dict(font=dict(size=16)),
    width=1500,
    height=1000,
    margin=dict(l=20, r=20, t=60, b=20)
    )

fig2.show()

#### The top 5 countries having the best average on life expectanvcy over the years are France, Sweeden, Iceland, Japan and Switzerland. The bottom 5 countries having the worrt average on life expectancy over the years are Sierra Leone, Malawi, Angola, Central African Republic and Lesotho.We can notice that the top 5 countries have an increasing on life expectancy (from 81-88) but it stabilises from 2009, whereas for the bottom 5 we can see a considerable growth (from 39 to 51 for certain countries). We can also notice that the top 5 countries belong to the northern hemisphere unlike the bottom 5 that are African Countries.

### 3. Violin Plot for Life Expectancy by Continent 

In [118]:
fig3 = px.violin(df, x='Continent', y='Life expectancy ', color='Continent',box=True,title='Violin Plot for Life Expectancy and Population by Continent')
# Update layout for better readability
fig3.update_layout(
    xaxis_title='Continent',
    yaxis_title='Life Expectancy',
    legend_title='Continent',
    title_font=dict(size=24),
    font=dict(size=15),
    legend=dict(font=dict(size=16)),
    width=1500,
    height=1000,
    margin=dict(l=20, r=20, t=60, b=20),
)

fig3.show()

#### We can conclude that Africa is the continent where life expectancy is low, so the authorities should concentrate on this continent. The second continent having a low average is Asia.

### 4. Pie chart for the distribution of countries by Status

In [119]:
fig4 = px.pie(
    df, 
    names='Status', 
    title='Distribution of Countries by Development Status',
    color='Status',  # Assigning colors based on the 'Status' column
    color_discrete_sequence=['#67001F', '#F4A582'],
    hole=0.4, 
)

# Update layout for better readability
fig4.update_layout(
    width=1500,
    height=1000,
    legend_title_text='Development Status',
    legend=dict(font=dict(size=16)),
    title_font=dict(size=24),
    font=dict(size=15)
)

fig4.show()

#### This dataset contains 17.4% of develoved countries and 82.6% of developping countries. This is a good ratio because usually it is developpping countries that have less life expectancy.

### 5. Comparing the GDP from 2000 to 2015 by the status


In [120]:
df_gdp_avg=df.groupby(['Year', 'Status'])['GDP'].mean().reset_index()


In [136]:
fig5 = px.bar(df_gdp_avg, x='Status', y='GDP', color='Status', title='Average GDP by Country Status Over Years',animation_frame='Year', barmode='group')
# Update layout for better readability
fig5.update_layout(
    xaxis_title='Year', 
    yaxis_title='Average GDP (USD)', 
    font=dict(size=15),
    title_font=dict(size=24),
    legend=dict(font=dict(size=16)),
    width=1200, 
    height=800,
    yaxis=dict(range=[0, 35000]))
fig5.show()

#### Over the years there is a big gap between the GDP of developed countries and developing countries taking into account the fact that  we have only 17% of developped countries.The gap between GDP and status is considerable. In 2000 we have a gap of 12 842 USD to 20 057 in 2014. There is an increase of GDP on both sides, but the increase is greater for developed countries.

### 6. Comparing the life expectancy from 2000 to 2015 by the status

In [122]:
fig6 = px.box(df, x='Status', y='Life expectancy ',title='Life Expectancy Distribution by Status (2000 to 2015)',animation_frame='Year',category_orders={'Year': sorted(df['Year'].unique())},labels={'Life expectancy': 'Life Expectancy', 'Status': 'Development Status'})
# Update layout for better readability
fig6.update_layout(
    xaxis_title='Development Status',
    yaxis_title='Life Expectancy',
    font=dict(size=15),
    title_font=dict(size=24),
    legend=dict(font=dict(size=16)),
    width=1300,
    height=1000,
    yaxis=dict(range=[35, 90])

)
fig6.show()

#### We can notice that there is an evolution of the life expactancy on both sides but for developing countries the values are more scattered than for developed countries where it is concentrated.

## Correlation study

### 7. Correlation map in order to study the columns that are influencing the life expectancy

In [123]:
numeric_columns = df.select_dtypes(include=['float64']).columns # Filtering the non-numeric columns in order to do a correlation
correlation_matrix = df[numeric_columns].corr()

In [124]:
fig7 = px.imshow(
    correlation_matrix,
    color_continuous_scale='GnBu',
    labels=dict(x='Features', y='Features', color='Correlation'),
    title='Correlation Heatmap',
    width=1500,
    height=1000,
)
# Update layout for better readability
fig7.update_layout(title_font=dict(size=24))
fig7.show()

#### From this correlation matrix we can see that Schooling, Income composition of ressources and BMI are highly corelated to life expectancy. It means that it influences the growth of life expectancy. Let's concentrate on the analysis of these columns

### 8. Correlation between schooling and life expectancy by continent 

In [125]:
fig8 = px.scatter(df, x='Schooling', y='Life expectancy ', trendline="ols",title='Scatter Plot of Schooling vs Life Expectancy',labels={'Schooling': 'Years of Schooling', 'Life expectancy ': 'Life Expectancy'})
# Update layout for better readability
fig8.update_layout(
    xaxis_title='Years of Schooling',
    yaxis_title='Life Expectancy',
    width=1500,
    height=1000,
    font=dict(size=15),
    title_font=dict(size=24),
    legend=dict(font=dict(size=16))
)
fig8.update_traces(
    line=dict(color='red', dash='solid'), 
    selector=dict(mode='lines')
)
fig8.show()

#### We can see that we have a cleary correlation line between schooling and Life expectancy. The more the years of schooling is the better is life expectancy.

### 9. Correlation between income ressources and life expectancy by continent in 2014

In [126]:
df_2014 = df[df['Year'] == 2014]

In [127]:
fig9 = px.scatter(df_2014, x='Income composition of resources', y='Life expectancy ', color='Continent',hover_name='Country', title='Income Composition vs Life Expectancy',labels={'Income composition of resources': 'Income Composition of Resources', 'Life expectancy': 'Life Expectancy'})
# Update layout for better readability
fig9.update_layout(
    xaxis_title='Income Composition of Resources',
    yaxis_title='Life Expectancy',
    width=1500,
    height=1000,
    font=dict(size=15),
    title_font=dict(size=24),
)
fig9.update_traces(marker=dict(size=8))

fig9.show()

#### Here we are only concentrating in 2014 because 2015 has many missing values, moreover it enables to clealy see the correlation. We can note that most contries from Africa have an Income composition resouces of 0.34-0.59 and a life expectancy of 48-68. For European countries we have a higher income composition of ressources and a better life expectancy.This explains the correlation between both the criterias, the more the income composition of ressources is the better is the life expectancy.

## Health study

### 10. Average BMI by continent

In [128]:
df_avg_bmi = df.groupby(['Continent', 'Year'])[' BMI '].mean().reset_index()


In [129]:
fig10 = px.bar(df_avg_bmi, x='Continent', y=' BMI ', color='Continent',animation_frame='Year',title='Average BMI Over the Years by Continent (2000-2014)',labels={' BMI ': 'Average BMI', 'Continent': 'Continent'},range_y=[df_avg_bmi[' BMI '].min(), df_avg_bmi[' BMI '].max()])
# Update layout for better readability
fig10.update_layout(
    xaxis_title='Continent',
    yaxis_title='Average BMI',
    width=1500,
    height=1000,
    font=dict(size=15),
    title_font=dict(size=24),
    legend=dict(font=dict(size=16))
)
fig10.show()

#### The body mass index (BMI) is a measure that uses your height and weight to work out if your weight is healthy.Compared to the other countinents Africa has the least average BMI score.This can explain the fact that it has a less life expectancy.Indeed a less BMI means that they are unhealthy. This can be caused by malnutrition and provoke earlier death.

### 11. Thinness between 1-19 years old accross countries

In [130]:
fig11 = px.choropleth(
    df,
    locations='Country',
    locationmode='country names',
    color=' thinness  1-19 years',
    hover_name='Country',
    color_continuous_scale=px.colors.sequential.Plasma,
    title='Thinness between 1-19 years old accross countries',
    template='plotly',
    animation_frame='Year',
    category_orders={'Year': sorted(df['Year'].unique())}
)
fig11.update_geos(
    resolution=110,
    showcoastlines=True,
    coastlinecolor="Black",
    showland=True,
    landcolor="white",
)
# Update layout for better readability
fig11.update_layout(
    width=1500,  
    height=1000,
    font=dict(size=15),
    title_font=dict(size=24)    
)
fig11.show()

#### In order to analyse our hypothesis made on the last visual we have made this map representing thinness between 1-19 years old accross countries.We can see that South Asian countries (like India, Pakistan) have the highest number of thinness between 1-19 year old. Compared to others, African countries are also having a relatively high number of thinness between 1-19 year old but we can also see that this has improved a little over the years. This can be an explaination for the BMI value.

### 12. Violin plot on Alcohol Consumption by continent

In [137]:
fig12 = px.violin(df, x='Continent', y='Alcohol', color='Continent',box=True,title='Violin plot on Alcohol Consumption by continent')
# Update layout for better readability
fig12.update_layout(
    xaxis_title='Continent',
    yaxis_title='Alcohol consumtion (in Liters)',
    legend_title='Continent',
    legend=dict(font=dict(size=16)),
    font=dict(size=15),
    title_font=dict(size=24),
    width=1500,
    height=1000,
    
)

fig12.show()

#### Europe is the continent where the alcohol consumtion is high compared to other continents. We can also see that it is highly spreaded.

### 13. Comparision on the evolution of HIV and Measles

In [132]:
df_hiv_measles = df.groupby('Year')[[' HIV/AIDS', 'Measles ']].sum().reset_index()

In [133]:
fig13 = sp.make_subplots(rows=2, cols=1, subplot_titles=['HIV/AIDS', 'Measles'])

bar_trace_hiv = go.Bar(x=df_hiv_measles['Year'], y=df_hiv_measles[' HIV/AIDS'], name='HIV/AIDS')
bar_trace_measles = go.Bar(x=df_hiv_measles['Year'], y=df_hiv_measles['Measles '], name='Measles')

fig13.add_trace(bar_trace_hiv, row=2, col=1)
fig13.add_trace(bar_trace_measles, row=1, col=1)
# Update layout for better readability
fig13.update_layout(
    title='Number of Deaths by HIV and Measles (2000-2015)',
    xaxis_title='Year',
    yaxis_title='Number of Deaths',
    width=1500,
    height=1000,
    font=dict(size=15),
    title_font=dict(size=24),
    legend=dict(font=dict(size=16))
)
fig13.update_xaxes(tickmode='linear')
fig13.update_xaxes( title_text='Year', row=2, col=1)

fig13.update_yaxes( title_text='Number of Deaths')
fig13.show()


#### Health wise, world wide we can see there is significant fall in the number of HIV and Measles. This decrease can explain the increase of life expectancy world wide.

### 14. Map on the evolution of Adult Mortality 

In [134]:
fig14 = px.choropleth(
    df,
    locations='Country',
    locationmode='country names',
    color='Adult Mortality',
    hover_name='Country',
    color_continuous_scale=px.colors.sequential.Plasma,
    title='Adult Mortality Across Countries',
    template='plotly',
    animation_frame='Year',
    category_orders={'Year': sorted(df['Year'].unique())}
)
fig14.update_geos(
    resolution=110,
    showcoastlines=True,
    coastlinecolor="Black",
    showland=True,
    landcolor="white",
)
# Update layout for better readability
fig14.update_layout(
    width=1500,  
    height=1000,
    font=dict(size=15),
    title_font=dict(size=24) 
)
fig14.show()

#### Over the years African countries and Asian countries have the highest number of adult deaths. This can be explained by health developpement, malnutitoin and also geo politic situation (we don't have more informations about this third point).

### 15. Map on the evolution of under five death to compare

In [135]:
fig15 = px.choropleth(
    df,
    locations='Country',
    locationmode='country names',
    color='infant deaths',
    hover_name='Country',
    color_continuous_scale=px.colors.sequential.Plasma,
    title='Under five mortality Across Countries',
    template='plotly',
    animation_frame='Year',
    category_orders={'Year': sorted(df['Year'].unique())}
)
fig15.update_geos(
    resolution=110,
    showcoastlines=True,
    coastlinecolor="Black",
    showland=True,
    landcolor="white",
)
# Update layout for better readability
fig15.update_layout(
    width=1500,  
    height=1000,
    font=dict(size=15),
    title_font=dict(size=24)  
)

fig15.show()

#### Over the years India and China, which are the most populated countries in the world, have a high number of under 5 year old death. We have already seen that India was also the country having the highest numbre of thinness between 1-19. These two elements can be correlated.

## Conclusion